doc/user/content/sql/create-source/webhook.md

```diff
@@ -123,3 +123,124 @@ Webhook sources apply the following limits to received requests:
+Webhook Destination.
+
+```
+CREATE SECRET segment_shared_secret AS 'abc123';
+```
+
+Using this shared key, Segment will sign each request and we can use the signature to determine if
+the request is legitmate.
+
+After defining a shared secret, we can create the source itself:
+
+```
+CREATE SOURCE my_segment_source IN CLUSTER my_cluster FROM WEBHOOK
+  BODY FORMAT JSON
+  INCLUDE HEADERS
+  CHECK (
+    WITH ( BODY BYTES, HEADERs, SECRET segment_shared_secret AS secret BYTES)
+    decode(headers->'x-signature', 'hex') = hmac(body, secret, 'sha1')
+  )
+```
+
+This creates a source called _my_segment_source_ and installs it in cluster named _my_cluster_.
+The source will have two columns, _body_ of type `jsonb` and _headers_ of type `map[text=>text]`.
+
+The `CHECK` statement defines how to validate each request. At the time of writing, Segment
+validates requests by signing them with an HMAC in the `X-Signature` request header. The HMAC is a
+hex-encoded SHA1 hash using the shared secret and request body. We can decode the signature using
+the [`decode`](/sql/functions/#decode) function, getting the raw bytes, and generate our own HMAC
+using the [`hmac`](/sql/functions/#hmac) function. If the two values are equal, then the request is
+legitimate!
+
+{{< note >}}
+
+For the latest information on Segment's Webhook Destination, please see their
+[documentation](https://segment.com/docs/connections/destinations/catalog/actions-webhook/).
+
+{{< /note >}}
+
+### Duplicated and Partial Events
+
+Given any number of conditions, e.g. a network hiccup, it's possible for your application to send
+an event more than once. If your event contains a unique ID you can de-duplicate these events
+using a [`MATERIALIZED VIEW`](/sql/create-materialized-view/) and the `DISCINCT ON` clause.
+
+```
+CREATE MATERIALIZED VIEW my_webhook_idempotent IN CLUSTER my_compute_cluster AS (
+  SELECT DISTINCT ON (body->>'unique_id') *
+  FROM my_webhook_source
+  ORDER BY id
+)
+```
+
+We can take this technique a bit further to handle partial events. Let's pretend our application
+tracks the completion of build jobs, and it sends us JSON objects with following structure.
+
+Key           | Value   | Optional? |
+--------------|---------|-----------|
+_id_          | `text`  | No
+_started_at_  | `text`  | Yes
+_finished_at_ | `text`  | Yes
+
+When a build job starts we receive an event containing _id_ and the _started_at_ timestamp. When a
+build finished, we'll receive a second event with the same _id_ but now a _finished_at_ timestamp.
+To merge these events into a single row, we can again use the `DISTINCT ON` clause.
+
+```
+CREATE MATERIALIZED VIEW my_build_jobs_merged IN CLUSTER my_compute_cluster AS (
+  SELECT DISTINCT ON (id) *
+  FROM (
+    SELECT
+      body->>'id' as id,
+      (body->>'started_at')::timestamptz as started_at,
+      (body->>'finished_at')::timestamptz as finished_at
```

Comment:
You should use `try_parse_monotonic_iso8601_timestamp` in this example. Otherwise, Materialize won't be able to leverage MFP. Normally I would say don't use features in private preview in an example but this is also in private preview 😅 

https://materialize.com/docs/sql/functions/pushdown/

Reply:
I understand this query doesn't have a temporal filter, but we should build the muscle memory to parse timestamps from JSON with this function.

Reply:
That's good to know! For now I added it as a note below this example, since the feature might not be enabled for user environments, how does that sound?